cocos/loop/LoopEvent.h

```diff
@@ -0,0 +1,58 @@
+/****************************************************************************
+Copyright (c) 2008-2010 Ricardo Quesada
+Copyright (c) 2009      Valentin Milea
+Copyright (c) 2010-2012 cocos2d-x.org
+Copyright (c) 2011      Zynga Inc.
+Copyright (c) 2013-2016 Chukong Technologies Inc.
+Copyright (c) 2017-2018 Xiamen Yaji Software Co., Ltd.
+
+http://www.cocos2d-x.org
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+****************************************************************************/
+
+
+#pragma once
+
+#include <string>
+namespace cocos2d
+{
+    namespace loop
+    {
+        class LoopEvent {
+        public:
+
+            LoopEvent(const std::string &msg) :message(msg) {}
+            LoopEvent(const std::string &msg, void *data) :message(msg), data(data) {}
+            LoopEvent(const LoopEvent &e) :eventName(e.eventName), message(e.message), data(e.data) {}
+            LoopEvent(LoopEvent && e) : eventName(e.eventName), message(e.message), data(e.data) {
+                e.data = nullptr;
+            }
+
+            void setName(const char *name) { eventName = name; }
+            const std::string &getName() const { return eventName; }
+        private:
+            std::string eventName{ "[noname]" };
+            std::string message;
+            void *data{ nullptr };
```

Comment:
Need to add `_` for class members, and it is better use `std::string _eventName = "[noname]"` as other codes.